### DISCUSSION TITLE
Implement Jmespath for Transform Payload

### DISCUSSION THREAD

**Discussion by Slembit** [directus/directus#14391] [Feature Requests]
Hi, really like the flows implementation! Great work.

I think JMESpath would be a real good transform "language" for the transform payload functionality.
Its usually very easy to implement and is quite effiecent.

https://jmespath.org/

is this something that could be considered?

**Comment by rijkvanzanten** [comment]
Transform language for payload functionality? I'm a little confused how/where/what this would be used for 🙂

**Reply by DeanKamali** [reply]
**Flows example** When making a third party API call we have very little control over the response body, with JMESpath we can filter / transform / pick single or multiple values, before we either (create / update / delete) values in the database or proceed with other flow logic. 
I believe it can cover most common use cases without the need to import third party libraries or run JavaScript in a sandbox.

**Reply by Slembit** [reply]
Couldn't have said it better than deankamali :)

**Reply by rijkvanzanten** [reply]
Thanks for explaining @DeanKamali! I think this would make a great addition to the "transform" operation we currently have. That has some (very basic) dot notation stuff included to construct a new JSON object from previous payloads, but doesn't have the same level of flexibility as JMES path

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
